pkg/frontend/querymiddleware/split_and_cache.go

```diff
@@ -127,6 +148,8 @@ func (s *splitAndCacheMiddleware) Do(ctx context.Context, req Request) (Response
 		return nil, err
 	}
 
+	s.metrics.queryResultCacheAttemptedCount.Add(float64(len(splitReqs)))
```

Comment:
This should be moved inside the `if isCacheEnabled` branch.

Reply:
Oops, make sense. Thanks!